composer/trainer/trainer.py

```diff
@@ -908,18 +979,47 @@ def __init__(
                 # that need to be moved onto GPU.
             optimizers = map_collection(optimizers, self._device.optimizer_to_device)
 
-        # Grad Accum
-        auto_grad_accum = _is_auto_grad_accum(grad_accum, device=self._device)
-        if auto_grad_accum and profiler:
-            raise ValueError("`grad_accum='auto'` is not compatible with the profiler. It is recommended to run "
-                             "a mini-run with `grad_accum='auto'` to identify the optimal grad_accum value and "
-                             'then manually specify that in a second run with profiler.')
-        grad_accum = _get_initial_grad_accum(grad_accum)
+        # Microbatching
+        # To support backwards compatability, we currently support both train_device_microbatch_size
+        # and grad_accum. If both are specified, we will use train_device_microbatch_size.
+        if train_device_microbatch_size:
+            using_device_microbatch_size = True
+            if grad_accum != 1:
+                raise ValueError(
+                    'Cannot use both train_device_microbatch_size and grad_accum. grad_accum is deprecated '
+                    'so it is recommended to use train_device_microbatch_size.')
+            grad_accum = None
+            auto_microbatching = _is_auto_microbatching(train_device_microbatch_size, device=self._device)
+            if auto_microbatching and profiler:
+                raise ValueError("`train_device_microbatch_size='auto'` is not compatible with the profiler. It is "
+                                 "recommended to run a mini-run with `train_device_microbatch_size='auto'` to identify "
+                                 'the optimal train_device_microbatch_size value and then manually specify that in a '
+                                 'second run with profiler.')
+            # If auto_microbatching is True, the microbatch size will be determined at when
+            # dataloader is specified.
+            train_device_microbatch_size = _get_initial_train_device_microbatch_size(train_device_microbatch_size, None)
+        elif grad_accum:
```

Comment:
is there any way you can just convert `grad_accum` to `train_device_microbatch_size` and then just use `train_device_microbatch_size` the rest of the way instead of having to check which one is set at multiple points in the code?

Reply:
Unfortunately, no since the batch size is not known a priori. A user could use an iterable which doesn't expose this information, so we can't convert eagerly.